experimental/Schemes/IdealSheaves.jl

```diff
@@ -576,6 +578,168 @@ function primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdea
   return result
 end
 
+## this should go to src/Rings/mpolyquo-localization.jl
+function minimal_primes(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdeal, <:MPolyLocalizedIdeal})
+  Q = base_ring(I)
+  R = base_ring(Q)
+  decomp = minimal_primes(saturated_ideal(I))
+  result = [ideal(Q, Q.(gens(b))) for b in decomp]
+  return result
+end
+
+
+function minimal_associated_points(I::IdealSheaf)
+  X = scheme(I)
+  OOX = OO(X)
+
+  charts_todo = copy(affine_charts(X))            ## todo-list of charts
+
+  associated_primes_temp = Vector{IdDict{AbsSpec, Ideal}}()  ## already identified components
+                                                  ## may not yet contain all relevant charts. but
+                                                  ## at least one for each identified component
+
+# run through all charts and try to match the components
+  while !iszero(length(charts_todo))
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                        ## supp(I) might not meet all components
+    components_here = minimal_primes(I(U))
+
+## run through all primes in MinAss(I(U)) and try to match them with previously found ones
+    for i in 1:length(components_here)
+      matches = match_on_intersections(X,U,components_here[i],associated_primes_temp,false)
+      nmatches = length(matches)
+
+      if nmatches == 0                             ## not found
+        add_dict = IdDict{AbsSpec,Ideal}()         ## create new dict
+        add_dict[U] = components_here[i]           ## and fill it
+        push!(associated_primes_temp, add_dict)
+      elseif nmatches == 1                         ## unique match, update it
+        component_index = matches[1]
+        associated_primes_temp[component_index][U] = components_here[i]
+      else                                                ## more than one match, form union
+        target_comp = pop!(matches)
+        merge!(associated_primes_temp[target_comp], associated_primes_temp[x] for x in matches)
+        for x in matches
+          deleteat!(associated_primes_temp,x)
+        end
+        associated_primes_temp[target_comp][U] = components_here[i]
+      end
+    end
+  end
+
+# fill the gaps arising from a support not meeting a patch
+  for U in affine_charts(X)
+    I_one = ideal(OOX(U),one(OOX(U)))
+    for i in 1:length(associated_primes_temp)
+      !haskey(associated_primes_temp[i],U) || continue
+      associated_primes_temp[i][U] = I_one
+    end
+  end
+
+# make sure to return ideal sheaves, not dicts
+  associated_primes_result = [IdealSheaf(X,associated_primes_temp[i],check=false) for i in 1:length(associated_primes_temp)]
+  return associated_primes_result
```

Comment:
Could also write this more compactly as
```suggestion
  return [IdealSheaf(X,p,check=false) for p in associated_primes_temp]
```

Reply:
It is written this way on purpose because some optional check might be added of which the code will then sit inbetween these two lines.